hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -748,7 +748,9 @@ public Target create(final TargetCreate c) {
     @Retryable(include = {
             ConcurrencyFailureException.class }, maxAttempts = Constants.TX_RT_MAX, backoff = @Backoff(delay = Constants.TX_RT_DELAY))
     public List<Target> create(final Collection<TargetCreate> targets) {
-        return targets.stream().map(this::create).collect(Collectors.toList());
+        final List<JpaTarget> targetList = targets.stream().map(JpaTargetCreate.class::cast).map(JpaTargetCreate::build)
+                .collect(Collectors.toList());
+        return new ArrayList<>(targetRepository.saveAll(targetList));
```

Comment:
I would still return `Collections.unmodifiableList` and adapt tests instead. The reason for that is after save operation you don't want to modify the resulting list, the only reason why we return this list is to get access to DB ids